packages/blob/src/index.js

```diff
@@ -70,3 +70,39 @@ export function isBlobURL( url ) {
 	}
 	return url.indexOf( 'blob:' ) === 0;
 }
+
+/**
+ * Downloads a file, e.g., a text or readable stream, in the browser.
+ * Appropriate for downloading smaller file sizes, e.g., < 5 MB.
+ *
+ * Example usage:
+ *
+ * ```js
+ * 	const fileContent = JSON.stringify(
+ * 		{
+ * 			"title": "My Post",
+ * 		},
+ * 		null,
+ * 		2
+ * 	);
+ * 	const fileName = 'file.json';
+ *
+ * 	downloadBlob( 'file.json', fileContent, 'application/json' );
+ * ```
+ *
+ * @param {string}   filename    File name.
+ * @param {BlobPart} content     File content (BufferSource | Blob | string).
+ * @param {string}   contentType File mime type.
+ */
+export function downloadBlob( filename, content, contentType ) {
+	const file = new window.Blob( [ content ], { type: contentType } );
```

Comment:
Can we add some validation to the parameters passed? 

Should at least a default mime type be defined in case nothing is passed? Will it work correctly if you pass the incorrect content type?

It's going to be a general purpose util available also outside the WordPress context.

Reply:
> Can we add some validation to the parameters passed?

Yes, good idea. Thanks!

> Should at least a default mime type be defined in case nothing is passed? Will it work correctly if you pass the incorrect content type?

I'll test this. Maybe `text/plain` could work.

It might be that a default mime-type might be incompatible with `content<any>` 🤔 so in that case we could require all the args.

Reply:
> Will it work correctly if you pass the incorrect content type?

I've tested with a fallback content type of `text/plain` and things like zips and images are still downloadable, and the browser somehow works it out.

However, `type` is an optional parameter and the default value is `""`, so I think we can can fallback to `""` and require the other args.